pkg/loop/internal/broker.go

```diff
@@ -34,28 +34,39 @@ func (a *atomicBroker) Accept(id uint32) (net.Listener, error) {
 	return a.load().Accept(id)
 }
 
-func (a *atomicBroker) Dial(id uint32) (conn *grpc.ClientConn, err error) {
-	return a.load().Dial(id)
+func (a *atomicBroker) DialWithOptions(id uint32, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
```

Comment:
can you explain to me how the plumb of the options is working here?

i expected you to need an upstream change to hashi corp, but i don't see that

Reply:
i see it now. i missed the replace on first reading

Reply:
ah shoot it does disrupt the /integration-tests modules too